java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -380,15 +380,27 @@ public void delete(final String rawPath) throws IOException {
      * @throws IOException If an IO exception has occurred.
      * @throws MantaClientHttpResponseException If a HTTP status code other than {@code 200 | 202 | 204} is encountered
      */
-    public void delete(final String rawPath,
-                       final MantaHttpHeaders requestHeaders) throws IOException {
-        Validate.notBlank(rawPath, "rawPath must not be blank");
+    public void delete(final String rawPath, final MantaHttpHeaders requestHeaders) throws IOException {
+        delete(rawPath, requestHeaders, config.getPruneEmptyParentDepth());
+    }
 
+    /**
+     * Deletes several directories in manta.
+     * @param paths - A list of paths to be deleted.
+     * @param requestHeaders - headers can be null.
+     * @throws IOException If a HTTP status code other than {@code 200 | 202 | 204} is encountered
+     */
+    void delete(final String rawPath, final MantaHttpHeaders requestHeaders, final Integer pruneDepth)
+            throws IOException {
+        Validate.notBlank(rawPath, "rawPath must not be blank");
         String path = formatPath(rawPath);
-        LOG.debug("DELETE {}", path);
-
-        httpHelper.httpDelete(path, requestHeaders);
-    }
+        if (pruneDepth == null) {
```

Comment:
It's good that you can handle the 0 case in `PruneEmpytParentDirectoryStrategy` but it's easier on the next maintainer if you make this check `pruneDepth == null || pruneDepth == 0` (since the fact that it isn't null means the unboxing will be safe)